pkg/docker/container.go

```diff
@@ -80,34 +90,32 @@ func EnsureContainer(ctx context.Context, ow *rpc.OutputWriter, cli *client.Clie
 
 	log.Infow("container not found; creating")
 
-	if opts.PullImageIfMissing {
-		out, err := cli.ImagePull(ctx, opts.ContainerConfig.Image, types.ImagePullOptions{})
+	switch opts.ImageStrategy {
```

Comment:
I've refactored this. We _don't_ want two different fixers:
- BuildImageFixer
- ContainerStartFixer

The reason is that a healthcheck could fail because the container exists but it's not started (e.g. testground-sidecar or our prometheus). The way it was implemented before, the image build would've ran in that case too.